### ISSUE TITLE
Feature Request - Better/Faster switching between different note folders

### ISSUE THREAD

**Issue by whollaus** [pbek/QOwnNotes#1676]
I have some ideas that would be helpful, at least for me:

* Add note folders to the toolbar
* Possibility to assign a key combination for each note folder.
* Operating system taskbar: Display note folder. Best to be able to choose the format freely. Currently it is "Note - QOwnNotes Version". But more helpful would be: "Note - Note Folder"
* Or another optical identifier in the task bar. (Similar display as for email programs of unread emails.)

**Comment by pbek** [comment]
Thank you for your suggestions!

> Add note folders to the toolbar

Isn't that overkill? You can already drag around the *Note folder panel* where you like.

> Possibility to assign a key combination for each note folder.

I tried that back then when implementing the shortcut settings, it had too many side effects. :disappointed: 

> Operating system taskbar: Display note folder. Best to be able to choose the format freely. Currently it is "Note - QOwnNotes Version". But more helpful would be: "Note - Note Folder"

Can you please elaborate on that one? I don't think I understand it yet.

> Or another optical identifier in the task bar. (Similar display as for email programs of unread emails.)

Which task bar are we talking about?

**Comment by whollaus** [comment]
> > Add note folders to the toolbar
> 
> Isn't that overkill? You can already drag around the _Note folder panel_ where you like.
yes, but the note folder panel is a drop down. So I don't see at first view, which folders are available. Additionally it is one click more to switch between the folders. This slows down the workflow. 

e.g. in Krusader you can set bookmarks. Each bookmark can be freely placed in the toolbar (if you want)

If it is too much effort, perhaps it could be solved with an addon? "Note Folder Button" with these options: Button Title and Note Folder select (Drop Down or plain text). Then we could insert as many buttons as we want.

Most exciting of course would be a new (additional) Note folder panel. Displayed as tabs (similar to firefox...; e.g. with definable colors per folder)

> > Operating system taskbar: Display note folder. Best to be able to choose the format freely. Currently it is "Note - QOwnNotes Version". But more helpful would be: "Note - Note Folder"
> 
> Can you please elaborate on that one? I don't think I understand it yet.
![Screenshot_20200406_103757](https://user-images.githubusercontent.com/4582757/78541472-f4719a00-77f5-11ea-8f3c-06e10891c1c1.jpg)

Currently "Todo - QOwnNotes 20.3.2" is displayed.
It would be better if the notes folder is also displayed (at least if there is more than one folder).
"Todo - Work - QOwnNotes 20.3.2" (Work is the folder)

Thunderbird, for example, makes it similar. So in the format "Order - Account - Thunderbird"

Basically, this would also be useful for different sessions. ("QOwnNotes --session Work" should display the session name)

**Comment by pbek** [comment]
> If it is too much effort, perhaps it could be solved with an addon? "Note Folder Button" with these options: Button Title and Note Folder select (Drop Down or plain text). Then we could insert as many buttons as we want.

Unfortunately you can't put custom actions in the toolbar, I had far too many unwanted side-effects with that.

But you can always use the keyboard shortcut to switch the note folder.

> Most exciting of course would be a new (additional) Note folder panel. Displayed as tabs (similar to firefox...; e.g. with definable colors per folder)

Switching to a MDI design would basically almost mean a rewrite of QOwnNotes. :grinning: 

> Currently "Todo - QOwnNotes 20.3.2" is displayed.
> It would be better if the notes folder is also displayed (at least if there is more than one folder).
> "Todo - Work - QOwnNotes 20.3.2" (Work is the folder)
> 
> Thunderbird, for example, makes it similar. So in the format "Order - Account - Thunderbird"
> 
> Basically, this would also be useful for different sessions. ("QOwnNotes --session Work" should display the session name)

Ah, you are talking about the title bar. Sounds like a good idea, can you please make a different feature request for that.

**Comment by whollaus** [comment]
I don't think the basic logic is changing. 
But the selection of folders is not done in the drop down. The folders are displayed as buttons next to each other. Controllable via the setting and everyone is happy.

Note order Display as:
* Drop down selection
* Buttons side by side
* Buttons below each other

![image10](https://user-images.githubusercontent.com/4582757/78546534-df990480-77fd-11ea-86fc-9b3f48b61cab.png)

**Comment by pbek** [comment]
With everyone, you mean "you"? 😆

**Comment by pbek** [comment]
But at least this would be doable in a finite amount of time. 😬

**Comment by whollaus** [comment]
> With everyone, you mean "you"? laughing

Yes, of course. :)

But I think this could be a very practical function. With improvements that could be made in the future.(Colors per folder, icon per folder,...).

**Comment by pbek** [comment]
Trickier than I thought to not get into a lot of regressions because of the new feature...

**Comment by pbek** [comment]
There now is a new release, could you please test it and report if it works for you?

**Comment by whollaus** [comment]
Perfect. Works great. 

Thank you very much for the quick implementation!

**Comment by pbek** [comment]
It was more work than I thought 😆

**Comment by whollaus** [comment]
That is why I am a web developer. A change like that wouldn't be a big problem. *g*

For me the function works perfectly. Is just much better now to switch between folders.
I prophesy: At some time, this will become the default setting. :-)

**Comment by pbek** [comment]
> A change like that wouldn't be a big problem.

Well, let's just say then you haven't got problem X and Y but instead you have problem A, B, C and D! :laughing: 
(I'm also a web-developer for >20y :grin:)

**Comment by pbek** [comment]
The "not introducing regressions" part is the hard one. QON has a (felt) million of settings and features that still need to work in all combinations.
